systemd-udevd/pkg.yaml

```diff
@@ -0,0 +1,127 @@
+        patch -p1 < /pkg/patches/0019-Adjust-for-musl-headers.patch
+        patch -p1 < /pkg/patches/0022-sd-event-Make-malloc_trim-conditional-on-glibc.patch
+        patch -p1 < /pkg/patches/0023-shared-Do-not-use-malloc_info-on-musl.patch
+        patch -p1 < /pkg/patches/0024-avoid-missing-LOCK_EX-declaration.patch
+        patch -p1 < /pkg/patches/0025-include-signal.h-to-avoid-the-undeclared-error.patch
+        patch -p1 < /pkg/patches/0026-undef-stdin-for-references-using-stdin-as-a-struct-m.patch
+        patch -p1 < /pkg/patches/0027-adjust-header-inclusion-order-to-avoid-redeclaration.patch
+        patch -p1 < /pkg/patches/0028-build-path.c-avoid-boot-time-segfault-for-musl.patch
+
+        # Own patches
+        patch -p1 < /pkg/patches/0001-chore-musl-headers-compatibility-for-udev.patch
+
+        meson setup build \
+          --buildtype=release \
+          -Dmode=release \
+          -Dlibdir=/usr/lib \
+          -Dversion-tag="{{ .systemd_version }}-talos" \
+          -Dselinux=enabled \
+          -Dblkid=enabled \
+          -Dkmod=enabled \
+          -Dopenssl=disabled \
+          -Dgshadow=false \
+          -Dutmp=false \
+          -Duserdb=false \
+          -Ddbus=disabled \
+          -Dglib=disabled \
+          -Dnss-mymachines=disabled \
+          -Dnss-myhostname=false \
+          -Dnss-resolve=disabled \
+          -Dnss-systemd=false \
+          -Dbacklight=false \
+          -Dbinfmt=false \
+          -Dbpf-framework=disabled \
+          -Dcoredump=false \
+          -Denvironment-d=false \
+          -Dfirstboot=false \
+          -Dhibernate=false \
+          -Dhostnamed=false \
+          -Dlink-udev-shared=false \
+          -Dlogind=false \
+          -Dmachined=false \
+          -Dnetworkd=false \
+          -Doomd=false \
+          -Dportabled=false \
+          -Dsysext=false \
+          -Danalyze=false \
+          -Dtimedated=false \
+          -Dtimesyncd=false \
+          -Dtmpfiles=false \
+          -Dpolkit=disabled \
+          -Dstatic-libudev=true \
+          -Dman=disabled \
+          -Dtests=false \
+          -Dwerror=false
+    build:
+      - |
+        meson compile -C build
+    install:
+      - |
+        meson install -C build --destdir /installroot
+
+        mkdir -p /rootfs/usr/bin
+        cp /installroot/usr/bin/udevadm /rootfs/usr/bin/
+
+        mkdir -p /rootfs/sbin
+        ln -s /usr/bin/udevadm /rootfs/sbin/udevadm
+        ln -s /usr/bin/udevadm /rootfs/sbin/systemd-udevd
+
+        mkdir -p /rootfs/usr/lib/
+        cp -r /installroot/usr/lib/udev /rootfs/usr/lib/
+        cp --preserve=links /installroot/usr/lib/libudev.* /rootfs/usr/lib/
+        rm -f /rootfs/usr/lib/udev/hwdb.d/README
+        rm -f /rootfs/usr/lib/udev/rules.d/{README,60-cdrom_id.rules,60-persistent-alsa.rules,60-persistent-v4l.rules,64-btrfs.rules,70-joystick.rules,70-mouse.rules,70-touchpad.rules,78-sound-card.rules,80-net-name-slot.rules,90-vconsole.rules,99-systemd.rules}
```

Comment:
Should we keep [90-iocost.rules](https://github.com/systemd/systemd/blob/d713c131a58b4c2b46ff0edc0fea26037edb2140/rules.d/90-iocost.rules) (untested) and should we fix hwdb or drop it? [hwdb errors](https://tpaste.us/W5YN) due to missing hwdb binary file which we have to build using systemd-hwdb

Reply:
I believe `hwdb` is required, otherwise it won't do proper stuff? I thought we had it with eudev

Reply:
We didn't, see my other comment where I looked at the difference in package contents. Anyway looking into that rn to see what it does except for HID devices and if we need to fix it or disable (via systemd build opts). iocost is also interesting, somewhat relevant to containers as per first Google search. If it can be useful to us then hwdb is apparently a must for it

Reply:
Okay, iocost fields are not populated in the default rules. However [it might be useful](https://tangchq74.github.io/IOCost_IEEE_TopPicks_23.pdf) and unless `IOCOST_SOLUTIONS` is set for a disk device it does nothing